### ISSUE TITLE
[Elastic Agent] Allow beats subprocess to define custom configuration.

### ISSUE THREAD

**Issue by ph** [elastic/beats#29860]
**Motivation**

**Use case 1:** Subprocess doesn't load specific custom beats configuration, this means that the beats will start with the default configuration that is defined in the **go-ucfg** structs from the source code. As an example, the **k8s processors** are enabled by default in the context of elastic-agent [when they should not](https://github.com/elastic/beats/pull/29809#issue-1100101810). 

**Use case 2:** Seccomp is on by default, with the default options and it's not currently possible to overrides these default and prevent [forking java-attacher](https://github.com/elastic/beats/issues/29631).

**Possible solution:**

Allow subprocess to load custom `{process_name}.elastic-agent.yml` when it starts, this could be added as part of the specs and the developer would be able to define a configuration that should be used.

**Questions:**

Is this introduce a security issue?

**Comment by ph** [comment]
I don't think that would introduce a bigger security issues, the files are owned by the right users, if you have already escalated privileges and you are able to edit the file, I think you can do more problems.

**Comment by simitt** [comment]
> Allow subprocess to load custom {process_name}.elastic-agent.yml when it starts, this could be added as part of the specs and the developer would be able to define a configuration that should be used.

@ph would this be extensible to user configuration eventually and how would that look like? 
https://github.com/elastic/apm-server/issues/6238 is an example where the default policy of standalone beats was missing a syscall used by the latest version of `glibc`. If `seccomp` filters are applied with a default policy, we either need to ensure that default policies cover for all supported architectures (hard to fully ensure) or allow user customization.

**Comment by ph** [comment]
@simitt It could be eventually be extended to user configuration.

I am looking a bit more into security and how we could lock down stuff, but I presume we could add a list of seccomp rules to the integration that the agent would aggregate and restart or apply it to the corresponding input.

Now can a user be able to change it, maybe, maybe it's an `opt-in` feature when managed in fleet.
